File: backend/src/main/java/com/senior/project/backend/domain/Submission.java
package com.senior.project.backend.domain;

import lombok.*;

import java.util.Date;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Generated
public class Submission {
	@Id
    private int id;
    private int taskId;
    private int artifactId;
    private UUID studentId;
	private Date submissionDate;
	private String comment;

    public void from(Submission submission) {
        this.id = submission.id;
        this.taskId = submission.taskId;
        this.artifactId = submission.artifactId;
        this.studentId = submission.studentId;
        this.submissionDate = submission.submissionDate;
        this.comment = submission.comment;
    }
}

File: frontend/crd/src/domain/Submission.ts
export interface SubmissionJSON {
  id: number;
  artifactId: number;
  taskId: number;
  studentId: string;
  submissionDate: Date;
  comment: string;
}

export class Submission {
  constructor(json: SubmissionJSON) {
    this.submissionId = json.id;
    this.artifactId = json.artifactId;
    this.taskId = json.taskId;
    this.studentId = json.studentId;
    this.submissionDate = new Date(json.submissionDate);
    this.comment = json.comment;
  }

  static make(
    artifactId: number,
    taskId: number,
    studentId: string,
    submissionDate: Date,
    comment: string
  ): Submission {
    return new Submission({
      id: 0,
      artifactId,
      taskId,
      studentId,
      submissionDate,
      comment
    });
  }

  static makeEmpty(): Submission {
    return new Submission({
      id: 0,
      artifactId: 1,
      taskId: 0,
      studentId: '',
      submissionDate: new Date(Date.now()),
      comment: ''
    });
  }

  hasFile(): boolean {
    return this.artifactId > 1;
  }

  submissionId: number;
  artifactId: number;
  taskId: number;
  studentId: string;
  submissionDate: Date;
  comment: string;
}
